src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -630,6 +631,214 @@ def assign_second_guardian_to_newborns(
+        other_race = "Multiracial or Other"
+        other_college_sims_idx = college_sims.loc[
+            college_sims["race_ethnicity"] == other_race
+        ].index
+        other_college_sims_with_single_female_reference_person_idx = (
+            other_college_sims_idx.intersection(
+                college_sims_with_single_female_reference_guardian_idx
+            )
+        )
+        other_college_sims_with_single_male_reference_person_idx = (
+            other_college_sims_idx.intersection(
+                college_sims_with_single_male_reference_guardian_idx
+            )
+        )
+        other_college_sims_with_partnered_reference_person_idx = (
+            other_college_sims_idx.intersection(
+                college_sims_with_partnered_reference_guardian_idx
+            )
+        )
+        # Pick guardian for college sims based on their guardian type and assign guardian
+        other_single_female_guardian_ids = self.randomness.choice(
+            other_college_sims_with_single_female_reference_person_idx,
+            choices=households_with_single_female_reference_person["reference_person_id"],
+            additional_key="other_single_female_guardian_ids",
+        )
+        other_single_male_guardian_ids = self.randomness.choice(
+            other_college_sims_with_single_male_reference_person_idx,
+            choices=households_with_single_male_reference_person["reference_person_id"],
+            additional_key="other_single_male_guardian_ids",
+        )
+        other_partnered_guardian_ids = self.randomness.choice(
+            other_college_sims_with_partnered_reference_person_idx,
+            choices=households_with_partners["reference_person_id"],
+            additional_key="other_partnered_guardian_ids",
+        )
+        pop.loc[
+            other_single_female_guardian_ids.index, "guardian_1"
+        ] = other_single_female_guardian_ids
+        # Should I use the original index of race_college_sims_with_single_female_reference_person_idx for lookup
+        pop.loc[
+            other_single_male_guardian_ids.index, "guardian_1"
+        ] = other_single_male_guardian_ids
+        pop.loc[
+            other_partnered_guardian_ids.index, "guardian_1"
+        ] = other_partnered_guardian_ids
+        # Assign guardian_2 for sims who's reference person has a partner
+        other_partner_ids = (
+            households_with_partners.reset_index()
+            .groupby(["reference_person_id"])["partner_id"]
+            .apply(np.random.choice)
+        )
+        pop.loc[other_partnered_guardian_ids.index, "guardian_2"] = pop["guardian_1"].map(
+            other_partner_ids
+        )
+
+        # Iterate through race/ethnicity and assign initial guardian
+        races = [
+            "AIAN",
+            "Asian",
+            "Black",
+            "Latino",
+            "NHOPI",
+            "White",
+        ]  # All races in state table excluding "Multiracial or other"
```

Comment:
Consider moving this variable to a constants file or top of this file and capitalizing the name `RACES`.